Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -310,17 +310,30 @@ PhysicalParticleContainer::AddGaussianBeam(Real x_m, Real y_m, Real z_m,
 }
 
 void
-PhysicalParticleContainer::AddPlasmaFromFile(const std::string s_f)
+PhysicalParticleContainer::AddPlasmaFromFile(const std::string s_f, amrex::Real q_tot)
 {
 #ifdef WARPX_USE_OPENPMD
-    openPMD::Series series = openPMD::Series(s_f, openPMD::AccessType::READ_ONLY);
+    openPMD::Series series=openPMD::Series(s_f,openPMD::AccessType::READ_ONLY);
     amrex::Print() << "openPMD standard version " << series.openPMD() << "\n";
-
     openPMD::Iteration& i = series.iterations[1];
-    amrex::Print()  << "File contains " << i.particles.size() << " specie(s):" << "\n";
-    for( auto const& ps : i.particles ) {
-        amrex::Print() << "\t" << ps.first << "\n";
-    }
+
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(i.particles.size()==1,"External file "
+                                     "should contain only one species\n");
+
+    std::pair<std::string,openPMD::ParticleSpecies> ps = *i.particles.begin();
+    amrex::Real p_m=ps.second["mass"][openPMD::RecordComponent::SCALAR].loadChunk<amrex::Real>().get()[0];
+    amrex::Real p_q=ps.second["charge"][openPMD::RecordComponent::SCALAR].loadChunk<amrex::Real>().get()[0];
+    int npart=ps.second["position"]["x"].getExtent()[0];
+    series.flush();
+
+    mass=p_m*PhysConst::mevpc2_kg;
+    charge=p_q*PhysConst::q_e;
+    Real weight=q_tot/charge/npart;
```

Comment:
Style:
```suggestion
    mass = p_m*PhysConst::mevpc2_kg;
    charge = p_q*PhysConst::q_e;
    Real weight = q_tot/charge/npart;
```

Logic: weight(ing) is also a per-particle, scalar attribute (such as mass and charge), unless guaranteed to be a constant record (component). Needs same assert as above, unless you want to infer the weighting from the total bunch charge you load (and then distribute equally?)?